src/main/java/org/takes/rs/Body.java

```diff
@@ -0,0 +1,254 @@
+/**
+ * The body of a response used by {@link RsWithBody}.
+ *
+ * @author Nicolas Filotto (nicolas.filotto@gmail.com)
+ * @version $Id$
+ * @since 0.32
+ */
+@SuppressWarnings("PMD.TooManyMethods")
+interface Body {
+    /**
+     * Gives an {@code InputStream} corresponding to the content of
+     * the body.
+     * @return The content of the body.
+     * @throws IOException in case the content of the body could not
+     *  be provided.
+     */
+    InputStream input() throws IOException;
+
+    /**
+     * Gives the length of the stream.
+     * @return The length of the stream.
+     * @throws IOException in case the length of the stream could not be
+     *  retrieved.
+     */
+    int length() throws IOException;
+
+    /**
+     * Content of a body based on an {@link java.net.URL}.
+     */
+    final class URL implements Body {
+
+        /**
+         * The {@link java.net.URL} of the content.
+         */
+        private final transient java.net.URL url;
+
+        /**
+         * Constructs an {@code URL} with the specified {@link java.net.URL}.
+         * @param content The {@link java.net.URL} of the content.
+         */
+        URL(final java.net.URL content) {
+            this.url = content;
+        }
+
+        @Override
+        public InputStream input() throws IOException {
+            return this.url.openStream();
+        }
+
+        @Override
+        public int length() throws IOException {
+            try (final InputStream input = this.url.openStream()) {
+                return input.available();
+            }
+        }
+    }
+
+    /**
+     * Content of a body based on a byte array.
+     */
+    final class ByteArray implements Body {
+
+        /**
+         * The content of the body in a byte array.
+         */
+        private final transient byte[] bytes;
+
+        /**
+         * Constructs an {@code ByteArray} with the specified byte array.
+         * @param content The content of the body.
+         */
+        ByteArray(final byte[] content) {
+            this.bytes = content.clone();
+        }
+
+        @Override
+        public InputStream input() {
+            return new ByteArrayInputStream(this.bytes);
+        }
+
+        @Override
+        public int length() {
+            return this.bytes.length;
+        }
+    }
+
+    /**
+     * The content of the body based on an {@link InputStream}.
+     */
+    final class Stream implements Body {
+
+        /**
+         * The content of the body in an InputStream.
+         */
+        private final transient InputStream stream;
+
+        /**
+         * The length of the stream.
+         */
+        private final transient int length;
+
+        /**
+         * Constructs an {@code Stream} with the specified {@link InputStream}.
+         * @param input The content of the body as stream.
+         */
+        Stream(final InputStream input) {
+            this.stream = input;
+            this.length = Body.Stream.length(input);
```

Comment:
@essobedo again, it's wrong. constructors must be code-free! see http://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html